core/src/lib.rs

```diff
@@ -51,6 +51,12 @@ pub type PermissionTokenDefinitionsMap =
 /// Type of `Sender<Event>` which should be used for channels of `Event` messages.
 pub type EventsSender = broadcast::Sender<Event>;
 
+/// Implemented for entities that produce events
+pub trait EventProducer {
+    /// Produce events.
+    fn produce_events(&self) -> Vec<Event>;
```

Comment:
Cool, `Into<Vec<Event>>` was a bit weird.

Reply:
and the code is moved from `data_model` into `iroha_core`